src/Workspaces/Core/Portable/Workspace/Solution/StateChecksums.cs

```diff
@@ -268,8 +267,7 @@ public DocumentStateChecksums(params object[] children) : base(WellKnownSynchron
 
             if (searchingChecksumsLeft.Remove(Text))
             {
-                // why I can't get text synchronously when async lazy support synchronous callback?
-                result[Text] = state;
+                result[Text] = await state.GetTextAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
💭 This seems strange. Are we sure the values are going to be treated as `SourceText` and not `TextDocumentState`? Do we have a test showing that the new value is correct (and possibly that the old value was wrong)?

Reply:
The `TextDocumentState` instance was previously replaced later on with the corresponding `SourceText` to avoid async call here. If we avoid allocating tasks via `ValueTask` making this method async is no longer a concern, so we can now store `SourceText` directly here.

Reply:
See https://github.com/dotnet/roslyn/pull/40091/files#diff-9914e58bcb07148a02f13b2cd9368212L23